common-service-core/src/main/java/com/ca/mfaas/message/core/AbstractMessageService.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.message.core;
+
+import com.ca.mfaas.message.storage.MessageTemplateStorage;
+import com.ca.mfaas.message.template.MessageTemplate;
+import com.ca.mfaas.message.template.MessageTemplates;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Abstract class which implements the {@link MessageService} interface.
+ * It creates messages from the file, validate and add them to the {@link MessageTemplateStorage}.
+ */
+@Slf4j
+public abstract class AbstractMessageService implements MessageService {
+
+    private final MessageTemplateStorage messageTemplateStorage;
+
+
+    /**
+     * Constructor that creates common messages and messages from file.
+     *
+     * @param messagesFilePath path to file with messages.
+     */
+    public AbstractMessageService(String messagesFilePath) {
+        messageTemplateStorage = new MessageTemplateStorage();
+        loadMessages(messagesFilePath);
+    }
+
+    /**
+     * Creates {@link Message} with key and list of parameters.
+     *
+     * @param key        of message in messages.yml file
+     * @param parameters for message
+     * @return {@link Message}
+     */
+    @Override
+    public Message createMessage(String key, Object... parameters) {
+        MessageTemplate messageTemplate = validateMessageTemplate(key);
+
+        try {
+            return Message.of(key, messageTemplate, parameters);
+        } catch (IllegalFormatConversionException | MissingFormatArgumentException exception) {
+            log.debug("Internal error: Invalid message format was used", exception);
+            messageTemplate = validateMessageTemplate(Message.INVALID_MESSAGE_TEXT_FORMAT);
+            return Message.of(key, messageTemplate, parameters);
+        }
+    }
+
+    /**
+     * Creates list {@link Message} with list of {@link Message}.
+     *
+     * @param key        of message in messages.yml file
+     * @param parameters list that contains arrays of parameters
+     * @return {@link Message}
+     */
+    @Override
+    public List<Message> createMessage(String key, List<Object[]> parameters) {
+        return parameters.stream()
+            .filter(Objects::nonNull)
+            .map(ob -> createMessage(key, ob))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Validate and add a {@link MessageTemplates} to the {@link MessageTemplateStorage}.
+     * @param messageTemplates the list of message templates
+     */
+    protected final void addMessageTemplates(MessageTemplates messageTemplates) {
+        validateMessageTemplates(messageTemplates);
+        messageTemplateStorage.addMessageTemplates(messageTemplates);
+    }
+
+    /**
+     * Validate {@link MessageTemplates} by checking that there are not occurrences of {@link MessageTemplate} with the same key.
+     * If there are, an exception is thrown.
```

Comment:
"Validate {@link MessageTemplates} by checking {@link MessageTemplate} occurrences with the same key. If {@link MessageTemplate} occur, an exception is thrown."